src/wallet/wallet.cpp

```diff
@@ -2550,14 +2534,66 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                     return false;
                 }
 
-                if (nFeeRet >= nFeeNeeded)
+                if (nFeeRet >= nFeeNeeded) {
+                    // Reduce fee to only the needed amount if we have change output to increase
+                    // Do not do this if fee was paid by payee
```

Comment:
I might be missing something, but instead of "Do not", the comment should read "TODO". Otherwise, it would imply that it is ok for the payee to lose coins whereas for the payer it is not.